service/serviceUtils.go

```diff
@@ -0,0 +1,211 @@
+package service
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/Informasjonsforvaltning/fdk-harvest-admin/config/env"
+	"github.com/Informasjonsforvaltning/fdk-harvest-admin/model"
+)
+
+func unmarshalAndValidateDataSource(bytes []byte) (*model.DataSource, error) {
+	var dataSource model.DataSource
+	err := json.Unmarshal(bytes, &dataSource)
+	if err != nil {
+		return nil, err
+	}
+
+	err = dataSource.Validate()
+	if err != nil {
+		return nil, err
+	}
+
+	return &dataSource, nil
+}
+
+func dataTypeToMessageKey(dataType model.DataTypeEnum) (*string, error) {
+	switch dataType {
+	case model.Concept:
+		msgKey := messageKey("concept")
+		return &msgKey, nil
+	case model.Dataset:
+		msgKey := messageKey("dataset")
+		return &msgKey, nil
+	case model.InformationModel:
+		msgKey := messageKey("informationmodel")
+		return &msgKey, nil
+	case model.DataService:
+		msgKey := messageKey("dataservice")
+		return &msgKey, nil
+	case model.PublicService:
+		msgKey := messageKey("publicservice")
+		return &msgKey, nil
+	}
+
+	return nil, errors.New(string(dataType) + " is not a valid data type")
```

Comment:
Har du vurdert `fmt.Errorf`? https://pkg.go.dev/fmt#Errorf

```suggestion
	return nil, fmt.Errorf("%s is not a valid data type", dataType)
```

Reply:
smooth :+1: 
e1001a9